server/server.js

```diff
@@ -49,7 +54,7 @@ app.use((req, res) => {
       RoutingContext(renderProps)
     );
 
-    const initialState = store.getState();
+    const initialState = authToken ? store.getState() : {}; // get token from request
```

Comment:
Thinking is state should not be set at all without a valid token. Otherwise, if we have set initial state, would it be possible for someone to access it?